cdm/core/src/main/java/ucar/nc2/filter/Classifier.java

```diff
@@ -46,25 +74,64 @@ public int[] classifyDoubleArray(Array arr) {
     return classifiedArray;
   }
 
-
-
   /** for a single double */
   public int classifyArray(double val) {
     if (val >= 0) {
       classifiedVal = 1;
     } else {
       classifiedVal = 0;
     }
-
     return classifiedVal;
   }
 
+  public int classifyArrayAttribute(double val) {
+    for (int[] rule : rules) {
+      if (val > rule[0] && val <= rule[1] + ucar.nc2.util.Misc.defaultMaxRelativeDiffFloat) {
```

Comment:
minor style comment: I would add an import statement for 
```
import ucar.nc2.util.Misc;
```
so that this can just be `Misc.defaultMaxRelativeDiffFloat`